app/models/floating_ip.rb

```diff
@@ -7,18 +7,14 @@ class FloatingIp < ApplicationRecord
   # TODO(lsmola) NetworkManager, once all providers use network manager rename this to
   # "ManageIQ::Providers::NetworkManager"
   belongs_to :ext_management_system, :foreign_key => :ems_id, :class_name => "ManageIQ::Providers::BaseManager"
-  # TODO(lsmola) NetworkManager, remove when all providers share the new network architecture
+  # Not going through network_port because of Amazon, old EC2 way allows to associate public Ip to instance, without
+  # any network_port used
```

Comment:
So, ultimately, the `belongs_to :vm` association should go away and is only used by the amazon provider?
Then how about moving the association to `Amazon::NetworkManager::FloatingIp` ?
Or what do you mean by 

> since this is used e.g. in tables